src/components/Identity/identityUtil.js

```diff
@@ -0,0 +1,61 @@
+import { AUTH_STATES } from "./constants";
+import isObject from "../../utils/isObject";
+
+const validateCustomerIDs = customerIDs => {
+  if (!isObject(customerIDs)) {
+    throw new Error(
+      "Invalid customer ID format. Each namespace should be an object"
+    );
+  }
+  Object.keys(customerIDs).forEach(customerID => {
+    if (!isObject(customerIDs[customerID])) {
+      throw new Error(
+        "Invalid customer ID format. Each namespace should be an object"
+      );
+    }
+    if (!customerIDs[customerID].id) {
+      throw new Error(
+        "Invalid customer ID format. Each namespace object should have an id"
+      );
+    }
+  });
+};
+const normalizeCustomerIDs = customerIDs => {
+  const normalized = {};
+  Object.keys(customerIDs).forEach(customerID => {
+    const currentID = customerIDs[customerID];
+    normalized[customerID] = { ...currentID };
+    if (
+      currentID.authState !== AUTH_STATES.AUTHENTICATED ||
+      currentID.authState !== AUTH_STATES.LOGGED_OUT
+    ) {
+      normalized[customerID].authState = AUTH_STATES.UNKNOWN;
+    }
+  });
+  return normalized;
+};
+const serializeCustomerIDs = customerIDs => {
+  return Object.keys(customerIDs)
+    .reduce((serialized, customerID) => {
+      return `${serialized}${customerID}|${customerIDs[customerID].id}${
+        customerIDs[customerID].authState
+      }|`;
+    }, "")
+    .slice(0, -1);
+};
+
+const createHashFromString = string => {
```

Comment:
Can we use the crypto APIs here, instead of this helper?

@Aaronius

Reply:
Here's how to create a sha-256 hash in a cross-browser way:

```
const text = 'An obscure body in the S-K System, your majesty. The inhabitants refer to it as the planet Earth.';

function buffer2hex(buffer) {
  return Array.prototype.map.call(
    new Uint8Array(buffer),
    function (x) { return ('00' + x.toString(16)).slice(-2) }
  ).join('');
}

function textEncode(str) {
  if (window.TextEncoder) {
    return new TextEncoder('utf-8').encode(str);
  } else {
    // IE 11, which doesn't have TextEncoder
    var utf8 = unescape(encodeURIComponent(str));
    var result = new Uint8Array(utf8.length);
    for (var i = 0; i < utf8.length; i++) {
      result[i] = utf8.charCodeAt(i);
    }
    return result;
  }
}

function digestMessage(message) {
  const data = textEncode(message);
  const crypto = window.msCrypto || window.crypto;
  const result = crypto.subtle.digest('SHA-256', data);
  if (result.then) {
    return result;
  } else {
    // IE 11, whose result is a CryptoOperation object instead of a promise
    return new Promise(function(resolve, reject) {
      result.addEventListener("complete", function() {
        resolve(result.result);
      });
      result.addEventListener("error", function() {
        reject();
      })
    })
  }
}

digestMessage(text).then(function(digestValue) {
  // Should result in 6efd383745a964768989b9df420811abc6e5873f874fc22a76fe9258e020c2e1
  console.log("Digest value in hex", buffer2hex(digestValue));
});
```

Reply:
I don't fully understand what the hash is used for, but are we sure that SHA-256 is what we want to use vs, say, SHA-1 or SHA-512?

Reply:
Thanks @Aaronius.

I think it's used today to hash the customer ids and auth states passed in, and store that hash in the cookie, so if the customers calls us again with the same IDs, we don't try to sync them since we've seen those IDs before.

So I don't think it's necessary to use 256

Reply:
Thank you @Aaronius for code sample.
I have updated this PR.